### DISCUSSION TITLE
More detailed upgrade information

### DISCUSSION THREAD

**Discussion by dootyfree** [google/timesketch#2754] [Q&A, quick solutions, support]
Can a more detailed and scenario specific upgrade information be documented.  As I have broken my TS environment several times in the past performing upgrades.  For minor changes the standard pull works.  For major changes this does not work.

Thanks!

**Comment by jaegeral** [comment]
Hey @dootyfree thanks for starting that discussion.

What specific scenarios are you thinking about?

What did break your TS environment and how did you fix it?

If you have specific issues, we welcome opening bugs for them (especially if troubleshooting and triaging was already done so we can better understand where the issue is). As the installations are sometimes very custom and we are not able to maintain documentation without feedback and suggestions from the community.

**Comment by dootyfree** [comment]
Hey @jaegeral,

I really didn't have time to dive into the problems just started from scratch each time.  No biggie.  
My scenario I'm wondering about is related to the new switch to v2 default gui.

The docker-compose/config.env file has changed in latest update correct?  So when trying a pull, will pull with the old compose and you have a hybrid old/new docker setup that does not work.  

So the concern is do I need to update the docker-compose/config.env before pulling?

Any guidance would be great.

Thanks!

**Comment by jaegeral** [comment]
The config.env file has last changed in March:https://github.com/google/timesketch/blob/5c5b83df34eab65bca6b1acd24ce680c47c419da/docker/release/config.env#L4

So again, not knowing what version / latest commit you  are running and what concrete issue you are facing, it is impossible to help.

**Comment by dootyfree** [comment]
Thanks jaegaral,

I believe docker is the preferred installation method.  So let's assume this as the 'most common' install type.  We know there is a commit which happened recently to change default gui to v2, correct?  There is also a change in the docker-compose to facilitate this correct?

Not sure where the disconnect is.  

If I am running the standard docker install where v2 is still in beta and I want to update the 'latest' which has v2 as primary.  How would I accomplish this?  Being that the docker-compose has changed.  

Pre update docker machines:
nginx
opensearch
timesketch-web-v2
timesketch-web
postgres
timesketch-worker
redis

New default UI docker machines:
 timesketch-web
timesketch-web-legacy
 timesketch-worker
 opensearch
postgres
redis
nginx

**Comment by jaegeral** [comment]
https://timesketch.org/guides/admin/upgrade/ is how it should work.

Important to know, depending on how long you have not updated, you might need to upgrade your database schema (see upgrade doc on how to do it and do not skip the backup step).

**Comment by dootyfree** [comment]
Ok, yes understand.  So if my current docker-compose has no reference to the 'timesketch-web-legacy' machine, and I perform the steps outlined to upgrade.  How can the /legacy be functional as this docker machine is not pulled.

**Reply by jaegeral** [reply]
> Ok, yes understand. So if my current docker-compose has no reference to the 'timesketch-web-legacy' machine, and I perform the steps outlined to upgrade. How can the /legacy be functional as this docker machine is not pulled.

That means you are not running the latest. See https://github.com/google/timesketch/blob/5c5b83df34eab65bca6b1acd24ce680c47c419da/docker/release/docker-compose.yml#L17

**Comment by dootyfree** [comment]
Thanks yes, this is why I'm upgrading.  To get to the latest.  I must be
confused sorry.  This is quite circular.

Seems a pull/down/up does not get you to the 'latest'.

Apologies if Im misunderstanding.

**Comment by jaegeral** [comment]
So if your docker compose does not have a mention of the legacy, than of course it will not build a web-legacy docker container, so maybe update your docker compose file.

Did you do a `git pull` ?
